specifyweb/specify/tests/test_record_merging/test_resolve_record_merge_response.py

```diff
@@ -0,0 +1,63 @@
+from unittest import skip
+from django import http
+from specifyweb.specify.record_merging import FailedMergingException, resolve_record_merge_response
+from specifyweb.specify.tests.test_api import ApiTests
+
+
+class TestResolveRecordMergeResponse(ApiTests):
+
+    def test_no_error(self):
+        response = resolve_record_merge_response(lambda: None)
+        self._assertStatusCodeEqual(response, 204)
+        self._assertContentEqual(response, "")
+
+    def _raise_error(self, error):
+    
+        def _raise():
+            raise error
+        return _raise
+    
+    def test_failed_merging_exception(self):
+
+        not_found = http.HttpResponseNotFound("model_name: Agent does not exist.")
+        
+        response = resolve_record_merge_response(
+            self._raise_error(
+                FailedMergingException(not_found)
+                )
+            )
+        
+        self.assertEqual(response, not_found)
+
+    def test_silent_error(self):
+
+        response = resolve_record_merge_response(
+            self._raise_error(Exception("Trying to merge into itself!")),
+        )
+        
+        self._assertStatusCodeEqual(response, http.HttpResponseServerError.status_code)
+        self.assertTrue("Trying to merge into itself!" in response.content.decode())
+
+    @skip("The type comparison in record merging is incorrect")
+    def test_not_found_not_silent(self):
+        not_found = http.HttpResponseNotFound("model_name: Agent does not exist.")
```

Comment:
https://github.com/specify/specify7/blob/c4f8414a497965c4a8e07a31dab63439903f9aa8/specifyweb/specify/record_merging.py#L59

^ it should be

`elif type(error.args[0]) == http.HttpResponseNotFound:`

The `type(http.HttpResponseNotFound)` will be a meta class.